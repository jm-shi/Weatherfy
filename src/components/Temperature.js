import React from 'react';
import PropTypes from 'prop-types';

class Temperature extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      showFahrenheit: true
    };

    this.useFahrenheit = this.useFahrenheit.bind(this);
    this.useCelsius = this.useCelsius.bind(this);
  }

  useFahrenheit() {
    this.setState({ showFahrenheit: true }, () => {
      this.props.useFahrenheit(this.state.showFahrenheit);
    });
  }

  useCelsius() {
    this.setState({ showFahrenheit: false }, () => {
      this.props.useFahrenheit(this.state.showFahrenheit);
    });
  }

  render() {
    const { showFahrenheit } = this.state;

    return (
      <div>
        <form>
          <input
            type="radio"
            id="radio-f"
            value="F"
            checked={showFahrenheit}
            readOnly
          />
          <label htmlFor="radio-f" onClick={this.useFahrenheit}>
            Fahrenheit
          </label>
          <input
            type="radio"
            id="radio-c"
            value="C"
            checked={!showFahrenheit}
            readOnly
          />
          <label htmlFor="radio-c" onClick={this.useCelsius}>
            Celsius
          </label>
        </form>
      </div>
    );
  }
}

Temperature.propTypes = {
  useFahrenheit: PropTypes.func
};

module.exports = Temperature;
