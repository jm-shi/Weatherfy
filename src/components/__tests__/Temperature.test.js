import React from 'react';
import Temperature from '../Temperature.js';
import { shallow } from 'enzyme';

test('should render Temperature correctly', () => {
  const wrapper = shallow(<Temperature />);
  expect(wrapper).toMatchSnapshot();
});
