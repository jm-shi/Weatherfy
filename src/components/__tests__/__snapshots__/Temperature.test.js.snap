// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Temperature correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Temperature />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <input
          checked={true}
          id="radio-f"
          readOnly={true}
          type="radio"
          value="F"
        />
        <label
          htmlFor="radio-f"
          onClick={[Function]}
        >
          Fahrenheit
        </label>
        <input
          checked={false}
          id="radio-c"
          readOnly={true}
          type="radio"
          value="C"
        />
        <label
          htmlFor="radio-c"
          onClick={[Function]}
        >
          Celsius
        </label>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={true}
            id="radio-f"
            readOnly={true}
            type="radio"
            value="F"
          />,
          <label
            htmlFor="radio-f"
            onClick={[Function]}
          >
            Fahrenheit
          </label>,
          <input
            checked={false}
            id="radio-c"
            readOnly={true}
            type="radio"
            value="C"
          />,
          <label
            htmlFor="radio-c"
            onClick={[Function]}
          >
            Celsius
          </label>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": true,
            "id": "radio-f",
            "readOnly": true,
            "type": "radio",
            "value": "F",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fahrenheit",
            "htmlFor": "radio-f",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Fahrenheit",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "id": "radio-c",
            "readOnly": true,
            "type": "radio",
            "value": "C",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Celsius",
            "htmlFor": "radio-c",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Celsius",
          "type": "label",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <input
            checked={true}
            id="radio-f"
            readOnly={true}
            type="radio"
            value="F"
          />
          <label
            htmlFor="radio-f"
            onClick={[Function]}
          >
            Fahrenheit
          </label>
          <input
            checked={false}
            id="radio-c"
            readOnly={true}
            type="radio"
            value="C"
          />
          <label
            htmlFor="radio-c"
            onClick={[Function]}
          >
            Celsius
          </label>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={true}
              id="radio-f"
              readOnly={true}
              type="radio"
              value="F"
            />,
            <label
              htmlFor="radio-f"
              onClick={[Function]}
            >
              Fahrenheit
            </label>,
            <input
              checked={false}
              id="radio-c"
              readOnly={true}
              type="radio"
              value="C"
            />,
            <label
              htmlFor="radio-c"
              onClick={[Function]}
            >
              Celsius
            </label>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": true,
              "id": "radio-f",
              "readOnly": true,
              "type": "radio",
              "value": "F",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Fahrenheit",
              "htmlFor": "radio-f",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Fahrenheit",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "id": "radio-c",
              "readOnly": true,
              "type": "radio",
              "value": "C",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Celsius",
              "htmlFor": "radio-c",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Celsius",
            "type": "label",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
